tf_encrypted/keras/losses.py

```diff
@@ -0,0 +1,53 @@
+"""TFE Keras loss function"""
+from tf_encrypted import get_protocol
+
+class Loss():
+  """Loss base class."""
+  def __init__(self,
+               fn,
+               **kwargs):
+
+    self.fn = fn
+    self._fn_kwargs = kwargs
+
+  def call(self, y_true, y_pred):
+    """Invokes the `LossFunctionWrapper` instance.
+    Args:
+      y_true: Ground truth values.
+      y_pred: The predicted values.
+    Returns:
+      Loss values per sample.
+    """
+    return self.fn(y_true, y_pred, **self._fn_kwargs)
+
+  def __call__(self, y_true, y_pred):
+    """Invokes the `Loss` instance.
+
+    Args:
+      y_true: Ground truth values.
+      y_pred: The predicted values.
+    """
+    losses = self.call(y_true, y_pred)
+    return losses
+
+
+class BinaryCrossentropy(Loss):
+  """Computes the cross-entropy loss between true l
+  abels and predicted labels.
+  """
+  def __init__(self):
+    super(BinaryCrossentropy, self).__init__(
+        binary_crossentropy)
+
+  def grad(self, y_true, y_pred):
+    return y_pred - y_true
+
+def binary_crossentropy(y_true, y_pred):
+
+  batch_size = y_true.shape.as_list()[0]
+  batch_size_inv = 1 / batch_size
+  out = y_true * get_protocol().log(y_pred)
```

Comment:
maybe we can't do it now but eventually (ie after refactoring) we should get rid of the `get_protocol` call and simply use `tfe.log(..)` instead (which will find the right protocol)

Reply:
Agreed